ddtrace/llmobs/_integrations/utils.py

```diff
@@ -397,6 +397,209 @@ def openai_set_meta_tags_from_chat(span: Span, kwargs: Dict[str, Any], messages:
+    input_messages = []
+    
+    if isinstance(input_data, str):
+        input_messages.append({"content": input_data, "role": "user"})
+    elif isinstance(input_data, list):
+        for m in input_data:
+            role = m.get("role", "user")
+            content = m.get("content", "")
+            if isinstance(content, list):
+                content_text = ""
+                for content_item in content:
+                    if isinstance(content_item, dict) and "text" in content_item:
+                        content_text += content_item.get("text", "")
+                    elif isinstance(content_item, str):
+                        content_text += content_item
+                    else:
+                        content_text += "[Unsupported content item type of {}]".format(type(content_item))
+            elif isinstance(content, str):
+                content_text = content
+            else:
+                content_text = "[Unsupported content type of {}]".format(type(content))
+            input_messages.append({"content": content_text, "role": role})
+    else:
+        input_messages.append({"content": "[Unsupported content type of {}]".format(type(input_data))})
+
+    # Add instructions to as an input system messages
+    if "instructions" in kwargs:
+        input_messages.insert(0, {"content": kwargs["instructions"], "role": "system"})
+    parameters = {
+        k: v for k, v in kwargs.items() if k not in ("model", "input", "api_key", "user_api_key", "user_api_key_hash")
+    }
+    span._set_ctx_items({INPUT_MESSAGES: input_messages, METADATA: parameters})
+    if span.error or not response:
+        span._set_ctx_item(OUTPUT_MESSAGES, [{"content": ""}])
+        return
+
+    # For streamed responses
+    if kwargs.get("stream"):
+        # To discuss: should I remove usage from streamed output message?
+        for item in response:
+            if "usage" in item:
+                item.pop("usage")
+        span._set_ctx_item(OUTPUT_MESSAGES, response)
+    # For non-streamed responses
+    # else:
+    output_data = _get_attr(response, "output", [])
+    output_messages = []
+    for output in output_data:
+        # Message or reasoning
+        if output.type == "message" or output.type == "reasoning":
+            role = _get_attr(output, "role", "")
+            content = _get_attr(output, "content", "")
+            summary = _get_attr(output, "summary", "")
+            encrypted_content = _get_attr(output, "encrypted_content", "")
+            if content:
+                message = {"content": "", "role": role}
+                for c in content:
+                    text = _get_attr(c, "text", "")
+                    message["content"] += text
+                output_messages.append(message)
+            if summary:
+                for s in summary:
+                    text = _get_attr(s, "text", "")
+                    output_messages.append({"content": text, "role": "reasoning"})
```

Comment:
Should we not combine all the summary text into one message? That is how you handle it in the streamed case.